### PR TITLE
Integration test with mocked client/transport

### PR DESCRIPTION
**Pull request by err508**

### CODE REVIEW

File: pathfinder/tests/test_blockchain_integration.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+from pathfinder.tests.fixtures import Mock
+from raiden_libs.blockchain import BlockchainListener
+
+
+def test_pfs_with_mocked_client(
+    generate_raiden_clients,
+    ethereum_tester,
+    contracts_manager: ContractManager,
+    blockchain_listener: BlockchainListener
+
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: remove new line

File: pathfinder/tests/test_blockchain_integration.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+):
+    """ instantiates a pathfinding service with mocked transport,
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: Capitalize pls

File: pathfinder/tests/test_blockchain_integration.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    listening and porcessing blockchain events created by mocked Raiden clients"""
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: typo in processing

File: pathfinder/tests/test_blockchain_integration.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    clients = generate_raiden_clients(7)
+    network_address = clients[0].contract.address
+
+    pathfinding_service = PathfindingService(
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: Why don't you use the PFS fixture that was added before? If not needed please remove the fixture.

File: pathfinder/tests/test_blockchain_integration.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+    #     p2_transferred_amount,
+    #     p2_fee
+    # )
+    # Topology:
+    #       /-------------\
+    # 0 -- 1 -- 2 -- 3 -- 4    5 -- 6
+    #  \-------/
+
+    channel_descriptions = [
+        (0, clients[0], 100, 20, 0.0010, 1, clients[1],  50, 10, 0.0015),  # capacities  90 --  60
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: I would remove the `clients[x]` here, and create them as local variables below to make the topology easier to read.

File: pathfinder/tests/test_blockchain_integration.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+        p1_deposit,
+        p1_transferred_amount,
+        p1_fee,
+        p2_index,
+        p2,
+        p2_deposit,
+        p2_transferred_amount,
+        p2_fee
+    ) in enumerate(channel_descriptions):
+        p1, p2 = token_network.channel_id_to_addresses[channel_id+1]
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: spaces around the `+`, strange that lint doesn't find that.

File: pathfinder/tests/test_blockchain_integration.py [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        p2_index,
+        p2,
+        p2_deposit,
+        p2_transferred_amount,
+        p2_fee
+    ) in enumerate(channel_descriptions):
+        pass
+
+    # there should be no channels
+    assert True
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: this doesn't do anything.

File: pathfinder/tests/test_blockchain_integration.py [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    pathfinding_service.token_network_listener.stop()
```
Comment (palango) [inline pathfinder/tests/test_blockchain_integration.py]: The shutdown should happen in the fixtures, just like the snipped I sent you.

File: pathfinder/tests/fixtures/network_service.py [hunk 9]
```diff
@@ -23,6 +24,17 @@ def forge_fee_signature(private_key: str, fee: float) -> bytes:
     return private_key_ecdsa.sign_recoverable(fee_msg)
 
 
+@pytest.fixture
```
Comment (pcppcp) [inline pathfinder/tests/fixtures/network_service.py]: use `yield` to return listener from the fixture and stop it manually, like here for example:

https://github.com/raiden-network/raiden-monitoring-service/blob/master/monitoring_service/test/fixtures/server.py#L54

File: pathfinder/tests/test_blockchain_integration.py (hunk 5 above)
Comment (pcppcp) [reply pathfinder/tests/test_blockchain_integration.py]: I think fixture for setting up the channel_descriptions/network would be a better choice.

Comment (palango) [reply pathfinder/tests/test_blockchain_integration.py]: Good point. But we can do that in a different PR.

File: pathfinder/tests/test_blockchain_integration.py (hunk 7 above)
Comment (err508) [reply pathfinder/tests/test_blockchain_integration.py]: I wrote that to have a reminder inside the code as well.

File: pathfinder/tests/test_blockchain_integration.py (hunk 8 above)
Comment (err508) [reply pathfinder/tests/test_blockchain_integration.py]: Done.

File: pathfinder/tests/fixtures/network_service.py (hunk 9 above)
Comment (err508) [reply pathfinder/tests/fixtures/network_service.py]: Done.

File: pathfinder/tests/test_blockchain_integration.py (hunk 5 above)
Comment (err508) [reply pathfinder/tests/test_blockchain_integration.py]: Done.

File: pathfinder/tests/test_blockchain_integration.py (hunk 4 above)
Comment (err508) [reply pathfinder/tests/test_blockchain_integration.py]: Done.

File: pathfinder/tests/test_blockchain_integration.py (hunk 7 above)
Comment (pcppcp) [reply pathfinder/tests/test_blockchain_integration.py]: just a TODO comment is ok. leaving assert here is confusing.

**Pull request merged**